### PR TITLE
Add constants for all strings

### PR DESCRIPTION
**Pull request by vlebourl**
Related Github issues: #110

### CODE REVIEW

**Comment by vlebourl** [comment]
any idea why that validation is failing?

**Comment by vlebourl** [comment]
i've run it on a couple of merged PR where it was successful, it fails as well. It looks like a bug on their side...

**Comment by iMicknl** [comment]
> any idea why that validation is failing?

The error is `ModuleNotFoundError: No module named 'voluptuous'` in HACS validation, however I am not sure why this fails.

File: custom_components/tahoma/binary_sensor.py [hunk 1]
```diff
@@ -89,13 +99,13 @@ def icon(self) -> Optional[str]:
...
+                return ICON_WATER
             else:
-                return "mdi:water-off"
+                return ICON_WATER
 
         icons = {
-            DEVICE_CLASS_GAS: "mdi:waves",
-            DEVICE_CLASS_RAIN: "mdi:weather-rainy",
+            DEVICE_CLASS_GAS: ICON_WAVES,
+            DEVICE_CLASS_RAIN: ICON_WATER_OFF,
```
Comment (iMicknl) [inline custom_components/tahoma/binary_sensor.py:108]: `ICON_WATER_OFF` is another icon?

File: custom_components/tahoma/const.py [hunk 2]
```diff
@@ -38,8 +38,9 @@
...
 }
 
-# TaHoma internal device states
 CORE_ON_OFF_STATE = "core:OnOffState"
 
-# Commands
 COMMAND_OFF = "off"
+COMMAND_ON = "on"
+
+ICON_WEATHER_WINDY = "mdi:weather-windy"
```
Comment (iMicknl) [inline custom_components/tahoma/const.py:46]: Tbh, if an ICON is used in two integrations I would just duplicate it for readability and manageability.

File: custom_components/tahoma/light.py [hunk 3]
```diff
@@ -100,14 +105,14 @@ def turn_on(self, **kwargs) -> None:
 
         if ATTR_BRIGHTNESS in kwargs:
             self._brightness = int(float(kwargs[ATTR_BRIGHTNESS]) / 255 * 100)
-            self.apply_action("setIntensity", self._brightness)
+            self.apply_action(COMMAND_SET_INTENSITY, self._brightness)
 
         elif ATTR_EFFECT in kwargs:
             self._effect = kwargs[ATTR_EFFECT]
-            self.apply_action("wink", 100)
+            self.apply_action(COMMAND_WINK, 100)
```
Comment (iMicknl) [inline custom_components/tahoma/light.py:112]:
```suggestion
            self.apply_action(self._effect, 100)
```

This is something we should have a look at. Is this not something that is passed dynamically? Not sure if self._effect is really needed as well.

File: custom_components/tahoma/lock.py [hunk 4]
```diff
@@ -46,29 +48,21 @@ def update(self):
             return
 
         self.controller.get_states([self.tahoma_device])
-        self._battery_level = self.tahoma_device.active_states["core:BatteryState"]
-        self._name = self.tahoma_device.active_states["core:NameState"]
         if (
-            self.tahoma_device.active_states.get("core:LockedUnlockedState")
+            self.tahoma_device.active_states.get(CORE_LOCKED_UNLOCKED_STATE)
             == STATE_LOCKED
```
Comment (iMicknl) [inline custom_components/tahoma/lock.py:53]: `STATE_LOCKED` is a constant from Home Assistant, not from the TaHoma component. Do we want to use those? My preference would be to have a specific one from the component, to keep it away from HA core.

File: custom_components/tahoma/binary_sensor.py [hunk 5]
```diff
@@ -89,13 +99,13 @@ def icon(self) -> Optional[str]:
         """Return the icon to use in the frontend, if any."""
         if self.device_class == DEVICE_CLASS_WATER:
             if self.is_on:
-                return "mdi:water"
+                return ICON_WATER
             else:
-                return "mdi:water-off"
+                return ICON_WATER
```
Comment (iMicknl) [inline custom_components/tahoma/binary_sensor.py:104]:
```suggestion
                return ICON_WATER_OFF
```

File: custom_components/tahoma/binary_sensor.py (hunk 1 above)
Comment (vlebourl) [reply custom_components/tahoma/binary_sensor.py:108]: hmm not sure what I did there, probably a bad copy and paste

File: custom_components/tahoma/const.py (hunk 2 above)
Comment (vlebourl) [reply custom_components/tahoma/const.py]: done

File: custom_components/tahoma/lock.py (hunk 4 above)
Comment (vlebourl) [reply custom_components/tahoma/lock.py]: done

File: custom_components/tahoma/binary_sensor.py (hunk 1 above)
Comment (vlebourl) [reply custom_components/tahoma/binary_sensor.py]: fixed

**Comment by vlebourl** [comment]
That was all because of #128...

**Pull request merged**